### PR TITLE
Simplify `YAMLEncoder` by replacing methods required by protocol with generic methods

### PR DESCRIPTION
**Pull request by norio-nomura**

### CODE REVIEW

File: Sources/Yams/Representer.swift [hunk 1]
```diff
@@ -192,38 +200,71 @@ extension Optional: NodeRepresentable {
...
+extension YAMLEncodable where Self: ScalarRepresentable {
+    public func box() -> Node {
+        return .scalar(represented())
+    }
+}
+
+extension Bool: YAMLEncodable {}
+extension Data: YAMLEncodable {}
+extension Decimal: YAMLEncodable {}
+extension Int: YAMLEncodable {}
```
Comment (jpsim) [inline Sources/Yams/Representer.swift]: Can we do `extension BinaryInteger: YAMLEncodable` or `extension Numeric: YAMLEncodable`?

File: Sources/Yams/Representer.swift [hunk 2]
```diff
@@ -192,38 +200,71 @@ extension Optional: NodeRepresentable {
...
+extension Int16: YAMLEncodable {}
+extension Int32: YAMLEncodable {}
+extension Int64: YAMLEncodable {}
+extension UInt: YAMLEncodable {}
+extension UInt8: YAMLEncodable {}
+extension UInt16: YAMLEncodable {}
+extension UInt32: YAMLEncodable {}
+extension UInt64: YAMLEncodable {}
+extension URL: YAMLEncodable {}
+extension String: YAMLEncodable {}
```
Comment (jpsim) [inline Sources/Yams/Representer.swift]: Can we do `extension StringProtocol: YAMLEncodable`?

File: Sources/Yams/Representer.swift (hunk 1 above)
Comment (norio-nomura) [reply Sources/Yams/Representer.swift]: Swift compiler does not allow writing extension of protocol to adding conformance.
```
Yams/Sources/Yams/Representer.swift:235:1: error: extension of protocol 'BinaryInteger' cannot have an inheritance clause
extension BinaryInteger: YAMLEncodable {
^                        ~~~~~~~~~~~~~
```

File: Sources/Yams/Representer.swift (hunk 2 above)
Comment (norio-nomura) [reply Sources/Yams/Representer.swift]: Same with `BinaryInteger`.
```
Yams/Sources/Yams/Representer.swift:250:1: error: extension of protocol 'StringProtocol' cannot have an inheritance clause
extension StringProtocol: YAMLEncodable {
^                         ~~~~~~~~~~~~~
```

File: Sources/Yams/Representer.swift (hunk 1 above)
Comment (jpsim) [reply Sources/Yams/Representer.swift]: Ah, right I forgot.

**Comment by jpsim** [comment]
👍

**Pull request merged**